lib/removePipelineExtras.js

```diff
@@ -0,0 +1,32 @@
+'use strict';
+var Cesium = require('cesium');
+var defined = Cesium.defined;
+
+module.exports = removePipelineExtras;
+
+function removePipelineExtras(gltf) {

```

Comment:
> When I'm adding the extras objects though, I would need some kind of separate reference for which objects are glTF objects, right?

I don't think so.  What exactly do you mean? You would recursively walk objects and add `extras`.  Recursion would stop if you reach a non-object property or an `extras` object.

> And just checking, is the removePipelineExtrasSpec file I added sufficient for unit testing (in this case)?

For the unit tests, I would create a glTF asset with all possible object properties (one of each), and have object properties

1. with no `extras` - verify that it still doesn't have `extras`

2. with an empty `extras` - verify that it is not removed.

3. with just `extras._pipeline` - verify that it is all removed.

4. with `extras` and an extras sub-property and `extras._pipeline` - verify that only `_pipeline` is removed.

Note that 2. is an interesting case that we didn't consider yet where an input asset has an `extras : {}` and _we should not remove it_ in case the creator of the asset wanted it to be there for use later.  So perhaps the way to handle this is to add a flag to the `_pipeline` object that says `deleteExtras`.  It will be true if `extras` was not originally defined.